// jest-cli/.haste_cache/preprocess-cache/armory_e724769052f12ac712b1a6202091c33a.jsx
'use strict'; // istanbul ignore next
var _createClass = (function () {function defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ('value' in descriptor) descriptor.writable = true;Object.defineProperty(target, descriptor.key, descriptor);}}return function (Constructor, protoProps, staticProps) {if (protoProps) defineProperties(Constructor.prototype, protoProps);if (staticProps) defineProperties(Constructor, staticProps);return Constructor;};})(); // istanbul ignore next
var _get = function get(_x, _x2, _x3) {var _again = true;_function: while (_again) {var object = _x, property = _x2, receiver = _x3;desc = parent = getter = undefined;_again = false;if (object === null) object = Function.prototype;var desc = Object.getOwnPropertyDescriptor(object, property);if (desc === undefined) {var parent = Object.getPrototypeOf(object);if (parent === null) {return undefined;} else {_x = parent;_x2 = property;_x3 = receiver;_again = true;continue _function;}} else if ('value' in desc) {return desc.value;} else {var getter = desc.get;if (getter === undefined) {return undefined;}return getter.call(receiver);}}}; // istanbul ignore next
function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { 'default': obj };} // istanbul ignore next
function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError('Cannot call a class as a function');}} // istanbul ignore next
function _inherits(subClass, superClass) {if (typeof superClass !== 'function' && superClass !== null) {throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass);}subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } });if (superClass) subClass.__proto__ = superClass;}var _faction_select = require('./faction_select');var _faction_select2 = _interopRequireDefault(_faction_select);var _storesArmory_store = require('./stores/armory_store');var _storesArmory_store2 = _interopRequireDefault(_storesArmory_store);var _smurfy_scraper = require('./smurfy_scraper');var _smurfy_scraper2 = _interopRequireDefault(_smurfy_scraper);var React = require('react');var Weapon = require('./weapon');var 
Armory = (function (_React$Component) {

  function Armory(props) {_classCallCheck(this, Armory);
    _get(Object.getPrototypeOf(Armory.prototype), 'constructor', this).call(this, props);

    // pre-bind
    this.update_weapon_list = this.update_weapon_list.bind(this);

    var store_data = _storesArmory_store2['default'].get_new_data();
    this.state = { 
      weapons_list: store_data.weapons_list[store_data.selected_faction] };}_inherits(Armory, _React$Component);_createClass(Armory, [{ key: 'update_weapon_list', value: 




    function update_weapon_list() {
      var store_data = _storesArmory_store2['default'].get_new_data();
      this.setState({ 
        weapons_list: store_data.weapons_list[store_data.selected_faction] });} }, { key: 'componentDidMount', value: 



    function componentDidMount() {
      _storesArmory_store2['default'].addChangeListener(this.update_weapon_list);} }, { key: 'render', value: 


    function render() {
      var weapons = this.state.weapons_list.map(function (weapon, index, weapons_list) {
        return React.createElement(Weapon, { weapon: weapon, key: index });});


      var laser_weapons = weapons.select;

      return React.createElement('armory', null, 
      React.createElement('h2', null, 'Armory'), 


      React.createElement(_smurfy_scraper2['default'], null), 
      React.createElement(_faction_select2['default'], null), 
      weapons);} }]);return Armory;})(React.Component);




module.exports = Armory;
