// jest-cli/.haste_cache/preprocess-cache/cooldown_store_f04aef6477ae22f5b10f0063b9c77762.jsx
'use strict';Object.defineProperty(exports, '__esModule', { value: true }); // istanbul ignore next
var _createClass = (function () {function defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ('value' in descriptor) descriptor.writable = true;Object.defineProperty(target, descriptor.key, descriptor);}}return function (Constructor, protoProps, staticProps) {if (protoProps) defineProperties(Constructor.prototype, protoProps);if (staticProps) defineProperties(Constructor, staticProps);return Constructor;};})(); // istanbul ignore next
var _get = function get(_x, _x2, _x3) {var _again = true;_function: while (_again) {var object = _x, property = _x2, receiver = _x3;desc = parent = getter = undefined;_again = false;if (object === null) object = Function.prototype;var desc = Object.getOwnPropertyDescriptor(object, property);if (desc === undefined) {var parent = Object.getPrototypeOf(object);if (parent === null) {return undefined;} else {_x = parent;_x2 = property;_x3 = receiver;_again = true;continue _function;}} else if ('value' in desc) {return desc.value;} else {var getter = desc.get;if (getter === undefined) {return undefined;}return getter.call(receiver);}}}; // istanbul ignore next
function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { 'default': obj };} // istanbul ignore next
function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError('Cannot call a class as a function');}} // istanbul ignore next
function _inherits(subClass, superClass) {if (typeof superClass !== 'function' && superClass !== null) {throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass);}subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } });if (superClass) subClass.__proto__ = superClass;}var _events = require('events');var _app_dispatcher = require('../app_dispatcher');var _app_dispatcher2 = _interopRequireDefault(_app_dispatcher);var _constantsCooldown_constants = require('../constants/cooldown_constants');var _constantsCooldown_constants2 = _interopRequireDefault(_constantsCooldown_constants);var _constantsMap_constants = require('../constants/map_constants');var _constantsMap_constants2 = _interopRequireDefault(_constantsMap_constants);var _storesHeat_store = require('../stores/heat_store');var _storesHeat_store2 = _interopRequireDefault(_storesHeat_store);var _map_store = require('./map_store');var _map_store2 = _interopRequireDefault(_map_store);var _heatsink_store = require(
'./heatsink_store');var _heatsink_store2 = _interopRequireDefault(_heatsink_store);

var data = { 
  time_to_zero: 0, 
  cool_rate: 0 };


var CHANGE = 'COOLDOWN_STORE_UPDATED';

function update(new_data) {
  data = Object.assign(data, new_data);}


function recalculate_coolrate() {
  var heatsink_store_data = _heatsink_store2['default'].get_new_data();

  var internal_heatsink_cooldown_modifier;
  var external_heatsink_cooldown_modifier;

  if (heatsink_store_data.double_heatsinks) {
    internal_heatsink_cooldown_modifier = .2;
    external_heatsink_cooldown_modifier = .14;} else 
  {
    internal_heatsink_cooldown_modifier = .1;
    external_heatsink_cooldown_modifier = .1;}


  var internal_cooldown = internal_heatsink_cooldown_modifier * heatsink_store_data.internal_heatsinks;
  var external_cooldown = external_heatsink_cooldown_modifier * heatsink_store_data.external_heatsinks;

  var cool_rate = +(internal_cooldown + external_cooldown).toFixed(2);

  // Map-specific modifier
  var mapstore_data = _map_store2['default'].get_new_data();
  if (typeof mapstore_data.selected_map != 'undefined') {
    cool_rate = cool_rate * mapstore_data.selected_map.dissipation;}


  data.cool_rate = cool_rate;}


function update_time_to_zero() {
  var time_to_zero = +(_storesHeat_store2['default'].get_new_data().value / data.cool_rate).toFixed(2);
  data.time_to_zero = time_to_zero;}var 


CooldownStore = (function (_EventEmitter) {function CooldownStore() {_classCallCheck(this, CooldownStore);_get(Object.getPrototypeOf(CooldownStore.prototype), 'constructor', this).apply(this, arguments);}_inherits(CooldownStore, _EventEmitter);_createClass(CooldownStore, [{ key: 'get_new_data', value: 

    function get_new_data() {
      return data;} }, { key: 'emitChange', 


    // move this to store base class
    value: function emitChange() {
      this.emit(CHANGE);} }, { key: 'addChangeListener', 


    // move this to store base class
    value: function addChangeListener(callback) {
      this.on(CHANGE, callback);} }, { key: 'removeChangeListener', 


    // move this to store base class
    value: function removeChangeListener(callback) {
      this.removeListener(CHANGE, callback);} }]);return CooldownStore;})(_events.EventEmitter);



var _CooldownStore = new CooldownStore();exports['default'] = 

_CooldownStore;

_CooldownStore.dispatch_token = _app_dispatcher2['default'].register(function (payload) {
  var action_type = payload.action_type;

  switch (action_type) {
    case _constantsCooldown_constants2['default'].COOLDOWN_UPDATE:
      recalculate_coolrate();
      _CooldownStore.emitChange();
      break;
    case _constantsCooldown_constants2['default'].COOLDOWN_ETA_UPDATE:
      update_time_to_zero();
      _CooldownStore.emitChange();
      break;
    case _constantsMap_constants2['default'].CHANGE_MAP:
      _app_dispatcher2['default'].waitFor([_map_store2['default'].dispatch_token]);
      recalculate_coolrate();
      _CooldownStore.emit(CHANGE);
      break;}});module.exports = exports['default'];
