// jest-cli/.haste_cache/preprocess-cache/weapon_actions_87b1d988cdfe122553eceaf7cb5b093a.jsx
'use strict';Object.defineProperty(exports, '__esModule', { value: true }); // istanbul ignore next
function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { 'default': obj };}var _app_dispatcherEs6 = require('../app_dispatcher.es6');var _app_dispatcherEs62 = _interopRequireDefault(_app_dispatcherEs6);var _constantsWeapon_constants = require('../constants/weapon_constants');var _constantsWeapon_constants2 = _interopRequireDefault(_constantsWeapon_constants);var _storesWeapon_store = require(
'../stores/weapon_store');var _storesWeapon_store2 = _interopRequireDefault(_storesWeapon_store);


var WeaponActions = { 

  /**
   * @param {component} weapon
   */
  equip: function equip(weapon) {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_EQUIP, 
      weapon_props: weapon.props.weapon });}, 



  unequip: function unequip(index) {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_UNEQUIP, 
      index: index });}, 



  strip_all: function strip_all() {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_STRIP_ALL });}, 



  group_fire: function group_fire(group_id) {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_GROUP_FIRE, 
      group_id: group_id });}, 



  alpha_strike: function alpha_strike() {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_ALPHA });}, 



  toggleWeaponGroup: function toggleWeaponGroup(equipped_weapon_id, group_id) {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_TOGGLE_GROUP, 
      equipped_weapon_id: equipped_weapon_id, 
      group_id: group_id });}, 




  start_cooldown: function start_cooldown(equipped_weapon_id) {
    _app_dispatcherEs62['default'].dispatch({ 
      action_type: _constantsWeapon_constants2['default'].WEAPON_COOLDOWN, 
      equipped_weapon_id: equipped_weapon_id });}, 




  toggle_list_faction: function toggle_list_faction() {
    _app_dispatcherEs62['default'].dispatch({});} };exports['default'] = 




WeaponActions;module.exports = exports['default'];
